### PR TITLE
URI invalid login credentials error

### PR DESCRIPTION
**Pull request by yousef-soliman**
URI invalid login credentials error
refactor: the way of init Judge

### CODE REVIEW

File: src/judges/interfaces.ts [hunk 1]
```diff
@@ -1,3 +1,7 @@
+export interface IJudge {
```
Comment (amrsalama) [inline src/judges/interfaces.ts]: We already have a class name `Judge` I think this is confusing

File: src/Scraper/interfaces.ts [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
 export interface IScraperOptions {
   headless?: boolean;
-  judges?: {
-    Codeforces?: IJudgeCredentials;
-    UVa?: IJudgeCredentials;
-    URI?: IJudgeCredentials;
-  };
+  judges?: IJudge[];
+}
+export interface IJudges{
```
Comment (amrsalama) [inline src/Scraper/interfaces.ts]: I think this should not be an interface, let's create `constants` file in `Scraper` dir and add this map into it

File: src/Scraper/Scraper.ts [hunk 3]
```diff
@@ -13,17 +14,17 @@ export default class Scraper {
...
 
       const logins: Promise<void>[] = [];
-      if (judges?.Codeforces) {
-        logins.push(new Codeforces(browser).login(judges.Codeforces));
-      }
 
-      if (judges?.UVa) {
-        logins.push(new UVa(browser).login(judges.UVa));
-      }
+      const JUDGES: IJudges = {
```
Comment (amrsalama) [inline src/Scraper/Scraper.ts]: Let's move it to `constants` file, See comment below.

File: src/Scraper/Scraper.ts [hunk 4]
```diff
@@ -13,17 +14,17 @@ export default class Scraper {
...
+        Codeforces: Codeforces,
+        UVa: UVa,
+        URI: URI
+      };
 
-      if (judges?.URI) {
-        logins.push(new URI(browser).login(judges.URI));
-      }
+      judges?.forEach(judge => {
+        const Judge: any = JUDGES[judge.judge];
```
Comment (amrsalama) [inline src/Scraper/Scraper.ts]: Don't use `any`, use its type.

File: src/Scraper/Scraper.ts [hunk 5]
```diff
@@ -13,17 +14,17 @@ export default class Scraper {
...
+      const JUDGES: IJudges = {
+        Codeforces: Codeforces,
+        UVa: UVa,
+        URI: URI
+      };
 
-      if (judges?.URI) {
-        logins.push(new URI(browser).login(judges.URI));
-      }
+      judges?.forEach(judge => {
```
Comment (amrsalama) [inline src/Scraper/Scraper.ts]: Maybe we can use object destructuring here, something like:
```js
 judges?.forEach(({ judge }) => {
```
So we don't use this `judge.judge` part below.

File: src/judges/interfaces.ts [hunk 6]
```diff
@@ -1,3 +1,7 @@
...
+  judge: string;
```
Comment (amrsalama) [inline src/judges/interfaces.ts]: Agree with @kerolloz this could be something like `judge: 'Codeforces' | 'URI' | 'UVa'`;

**Comment by yousef-soliman** [comment]
@amrsalama @kerolloz  please review the new changes

File: src/Scraper/Scraper.ts (hunk 4 above)
Comment (yousef-soliman) [reply src/Scraper/Scraper.ts]: which type?

**Comment by amrsalama** [comment]
It will need more refactoring, maybe in another commit.

**Pull request merged**